plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/ui/LoginBrowser.java

```diff
@@ -0,0 +1,96 @@
+package com.google.cloud.tools.eclipse.appengine.login.ui;
+
+import java.net.URL;
+
+import org.eclipse.jface.dialogs.Dialog;
+import org.eclipse.jface.dialogs.IDialogConstants;
+import org.eclipse.jface.layout.GridDataFactory;
+import org.eclipse.jface.window.IShellProvider;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.browser.Browser;
+import org.eclipse.swt.browser.ProgressAdapter;
+import org.eclipse.swt.browser.ProgressEvent;
+import org.eclipse.swt.browser.TitleEvent;
+import org.eclipse.swt.browser.TitleListener;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Shell;
+
+public class LoginBrowser extends Dialog {
```

Comment:
Needs a class comment.  The class is specific for Google Login URLs (the constructor takes an arbitrary URL, so it's good to spell this out).  It's assumed that the OAuth flow will be using the `urn:ietf:wg:oauth:2.0:oob` callback (or are you using `urn:ietf:wg:oauth:2.0:oob:auto`?).